src/Uno.UI/UI/Xaml/Window.Android.cs

```diff
@@ -76,6 +80,57 @@ private static Window InternalGetCurrentWindow()
 
 		internal void RaiseNativeSizeChanged()
 		{
+#if __ANDROID_30__
+			var (windowBounds, visibleBounds, trueVisibleBounds) = Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.R
+				? GetVisualBounds()
+				: GetVisualBoundsLegacy();
+#else
+			var (windowBounds, visibleBounds, trueVisibleBounds) = GetVisualBoundsLegacy();
+#endif
+
+			ApplicationView.GetForCurrentView()?.SetVisibleBounds(visibleBounds);
+			ApplicationView.GetForCurrentView()?.SetTrueVisibleBounds(trueVisibleBounds);
+
+			if (Bounds != windowBounds)
+			{
+				Bounds = windowBounds;
+
+				RaiseSizeChanged(new WindowSizeChangedEventArgs(new Size(Bounds.Width, Bounds.Height)));
+			}
+		}
+
+#if __ANDROID_30__
+		private (Rect windowBounds, Rect visibleBounds, Rect trueVisibleBounds) GetVisualBounds()
+		{
+			var metrics = (ContextHelper.Current as Activity)?.WindowManager?.CurrentWindowMetrics;
+
+			var insetsTypes = WindowInsets.Type.SystemBars(); // == WindowInsets.Type.StatusBars() | WindowInsets.Type.NavigationBars() | WindowInsets.Type.CaptionBar();
+			var solidInsetsTypes = insetsTypes;
+			if (IsStatusBarTranslucent())
+			{
+				solidInsetsTypes &= ~WindowInsets.Type.StatusBars();
+			}
+			if (IsNavigationBarTranslucent())
+			{
+				solidInsetsTypes &= ~WindowInsets.Type.NavigationBars();
+			}
+
+			var insets = metrics.WindowInsets.GetInsets(insetsTypes).ToThickness();
+			var solidInsets = metrics.WindowInsets.GetInsets(solidInsetsTypes).ToThickness();
+
+			// The 'metric.Bounds' does not include any insets, so we remove the "solid" insets under which we cannot draw anything
+			var windowBounds = new Rect(default, ((Rect)metrics.Bounds).DeflateBy(solidInsets).Size).PhysicalToLogicalPixels();
+
+			// The visible bounds is the windows bounds on which we remove also
+			// [translucent only inset](all_insets - solid_insets_that_has_already_been_removed_from_all_insets)
+			var visibleBounds = windowBounds.DeflateBy(insets.Minus(solidInsets)).PhysicalToLogicalPixels();
+
+			return (windowBounds, visibleBounds, visibleBounds);
```

Comment:
I'd like to understand this better, the original distinction between VisibleBounds (public) and TrueVisibleBounds (internal) was introduced to avoid a breaking change in the semantics of VisibleBounds.

What is changing here w.r.t. the old code path - the value of VisibleBounds, or the value of TrueVisibleBounds?

Reply:
Discuss, as long as tests are passing, we agreed that it's correct